ibm_cloud_sdk_core/base_service.py

```diff
@@ -23,134 +23,72 @@
 from requests.structures import CaseInsensitiveDict
 from .version import __version__
 from .utils import has_bad_first_or_last_char, remove_null_values, cleanup_values
-from .iam_token_manager import IAMTokenManager
-from .icp4d_token_manager import ICP4DTokenManager
 from .detailed_response import DetailedResponse
 from .api_exception import ApiException
-
-try:
-    from http.cookiejar import CookieJar  # Python 3
-except ImportError:
-    from cookielib import CookieJar  # Python 2
+from .authenticators import Authenticator, BasicAuthenticator, BearerAuthenticator, CP4DAuthenticator, IAMAuthenticator, NoAuthAuthenticator
+from http.cookiejar import CookieJar
 
 # Uncomment this to enable http debugging
-# try:
-#    import http.client as http_client
-# except ImportError:
-#    # Python 2
-#    import httplib as http_client
+# import http.client as http_client
 # http_client.HTTPConnection.debuglevel = 1
 
+
 class BaseService(object):
-    BEARER = 'Bearer'
-    ICP_PREFIX = 'icp-'
-    APIKEY = 'apikey'
-    IAM_ACCESS_TOKEN = 'iam_access_token'
-    ICP4D_ACCESS_TOKEN = 'icp4d_access_token'
-    URL = 'url'
-    USERNAME = 'username'
-    PASSWORD = 'password'
-    IAM_APIKEY = 'iam_apikey'
-    IAM_URL = 'iam_url'
-    ICP4D_URL = 'icp4d_url'
-    APIKEY_DEPRECATION_MESSAGE = 'Authenticating with apikey is deprecated. Move to using Identity and Access Management (IAM) authentication.'
     DEFAULT_CREDENTIALS_FILE_NAME = 'ibm-credentials.env'
```

Comment:
We should no longer have any references to credential file-related things in BaseService.
The BaseService class will need to retrieve any properties related to the service (based on service name) so that it can obtain the service-level URL and disableSSL properties, but hopefully BaseService doesn't need to concern itself with any of those details other than just use a utility class to retrieve the properties.